apps/api/src/api/application-api.ts

```diff
@@ -32,9 +33,13 @@ import { failure } from '@/utils/results.js';
  */
 export const createApplication = async ({ user_id }: { user_id: string }) => {
 	const database = getDbInstance();
-	const service: ApplicationService = applicationService(database);
+	const applicationRepo: ApplicationService = applicationService(database);
+	const actionRepo: ActionService = actionService(database);
 
-	const result = await service.createApplication({ user_id });
+	const result = await applicationRepo.createApplication({ user_id });
+	if (result.success && result.data) {
+		await actionRepo.create(result.data);
```

Comment:
if this `create` fails, will it break the `actions` for the this application?

phrasing it a different way, will there a missing application state when there should be an entry with the state of `ApplicationStates.DRAFT` ?

I'm not too clear on how this project uses the state machine, but I imagine there needs to be valid state all the way through or there will be errors.

Reply:
Action creation is grouped in a transaction with the create step so failures should be managed:
https://github.com/Pan-Canadian-Genome-Library/daco/blob/feat/89-application-history-service/apps/api/src/service/application-service.ts#L44

Reply:
I see, so does it need to happen on line 41 also?

Reply:
No, that can be removed. I've taken this out and moved the related Create Action unit test out of API to the service test file